File: RoRClient/Commands/Editor/Other/LeaveEditorCommand.cs
using RoRClient.Commands.Base;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using RoRClient.Communication.DataTransferObject;
using RoRClient.Models.Session;
using RoRClient.Models.Game;

namespace RoRClient.Commands.Editor.Other
{
    public class LeaveEditorCommand : CommandBase
    {
        private Guid playerId;

        public LeaveEditorCommand(RoRSession session, MessageInformation message) : base(session, message)
        {
            playerId = message.GetValueAsGuid("playerId");
        }

        public override void Execute()
        {
            Boolean deleteEditorSession = false;
            Player player = EditorSession.GetInstance().GetPlayerById(playerId);

            // Wenn der Spieler Host ist oder kein anderer
            if (player == EditorSession.GetInstance().OwnPlayer || player.IsHost)
            {
                deleteEditorSession = true;
            }

            EditorSession.GetInstance().RemovePlayer(player);

            // Dann auch die GameSession löschen
            if (deleteEditorSession)
            {
                EditorSession.GetInstance().DeleteEditorSession();
            }
        }
    }
}

File: RoRServer/src/main/java/commands/editor/LeaveEditorCommand.java
package commands.editor;

import java.util.UUID;
import commands.base.CommandBase;
import communication.MessageInformation;
import models.game.Player;
import models.session.EditorSession;
import models.session.EditorSessionManager;
import models.session.RoRSession;

public class LeaveEditorCommand extends CommandBase {

	private UUID playerId;
	private boolean isHost;

	public LeaveEditorCommand(RoRSession session, MessageInformation messageInfo) {
		super(session, messageInfo);
		playerId = messageInfo.getValueAsUUID("playerId");
		isHost = messageInfo.getValueAsBoolean("isHost");
	}

	@Override
	public void execute() {
		Player player = session.getPlayerById(playerId);

		// Alle Spieler löschen wenn Spieler Host ist
		if (isHost) {
			EditorSessionManager.getInstance().removeEditorSession((EditorSession) session, true);
			// Nur Spieler löschen
		} else {
			session.removePlayer(player, true);
		}
	}
}
